### PR TITLE
1.Restricting EDIT/DELETE options 2.resolving the error for changes made in JASON fields

### PR DESCRIPTION
**Pull request by Snehal1894**
1) This will prevent uses from changing objects that are shared by everyone.
2) There are fields in forms that show different fields based on selection type.
When such fields are changed, their associated JSON fields also change, so comparing the old and updated fields was giving error  (like key not found).

### CODE REVIEW

**Comment by sergpolly** [comment]
@Snehal1894 - looks nice !
although I wonder if there is a better way to just commenting the buttons ... you're commenting them for all the users, right, including admins ? SO in order to actually remove something admins would need to go to the admin panel to curate database directly, right ?
maybe we could leave a message on the page somewhere - instead of those buttons - something like - "in order to edit/delete your records please contact your admin" ...

or even maybe leave the buttons edit/delete, but when pressed instead of editing/deleting - they'll show a meesage: "in order to edit/delete your records please contact your admin" - i.e. we'd leave the functionality there (maybe for the future) - what do you think about that ?

**Comment by Snehal1894** [comment]
> @Snehal1894 - looks nice !
> although I wonder if there is a better way to just commenting the buttons ... you're commenting them for all the users, right, including admins ? SO in order to actually remove something admins would need to go to the admin panel to curate database directly, right ?
> maybe we could leave a message on the page somewhere - instead of those buttons - something like - "in order to edit/delete your records please contact your admin" ...
> 
> or even maybe leave the buttons edit/delete, but when pressed instead of editing/deleting - they'll show a meesage: "in order to edit/delete your records please contact your admin" - i.e. we'd leave the functionality there (maybe for the future) - what do you think about that ?

Even if you do that, admins would still be needing to go the admin panel to make any changes.

File: cLIMS/organization/templates/detailConstructs.html [hunk 1]
```diff
@@ -7,9 +7,9 @@
  <script src="{% static "siteWide/js/show.js" %}"></script> 
 {% if construct %}
 <div class="bodyContent container">
-<a href="#">
+<!--<a href="#">
```
Comment (sergpolly) [inline cLIMS/organization/templates/detailConstructs.html:10]: i wonder what did this button even do before ? https://stackoverflow.com/questions/4855168/what-is-href-and-why-is-it-used

`href=#` goes scrolls on top of the pcurrent page without doing much... am I missing something ?

**Comment by sergpolly** [comment]
yes this is true, but with the last scenario:

>or even maybe leave the buttons edit/delete, but when pressed instead of editing/deleting - they'll show a meesage: "in order to edit/delete your records please contact your admin" - i.e. we'd leave the functionality there (maybe for the future) - what do you think about that ?

the html wouldn't change so it's a bit nicer for the users maybe - they used to see those buttons there...
That would require more work perhaps

File: cLIMS/organization/templates/detailConstructs.html (hunk 1 above)
Comment (Snehal1894) [reply cLIMS/organization/templates/detailConstructs.html:10]: Yes it just goes to the top of the page.
So when you click on edit/delete button.
If you check the url , # is appended, and stays on the same page

**Comment by sergpolly** [comment]
for the lack of time - we need to use thios as is - to prevent users from "damaging" the database - we need to think of what to do next with it

**Pull request merged**